packages/components/src/hooks/useClearClosedEventsFromApolloCache.ts

```diff
@@ -0,0 +1,32 @@
+import { useApolloClient } from '@apollo/client';
+import { useEffect } from 'react';
+import type { EventListQuery } from '../types/generated/graphql';
+import { isEventClosed } from '../utils/eventUtils';
+
+/**
+ * Clear the Apollo cache from the events of the past
+ */
+export default function useClearClosedEventsFromApolloCache(
+  data?: EventListQuery
+) {
+  const { cache } = useApolloClient();
+  useEffect(() => {
+    const eventsOfPastIds = data?.eventList.data.reduce<string[]>(
+      (pastEventIds, event) =>
+        isEventClosed(event) ? [...pastEventIds, event.id] : pastEventIds,
+      []
+    );
+    if (eventsOfPastIds?.length) {
+      eventsOfPastIds?.forEach((eventId) =>
+        cache.evict({ id: `EventDetails:${eventId}` })
+      );
+      cache.gc();
+      // eslint-disable-next-line no-console
+      console.info(
+        // eslint-disable-next-line no-console
```

Comment:
Needless second `eslint-disable-next-line no-console` unless there's something else going on than simply "yarn lint" under packages/components (I tried removing this second eslint-disable and running "yarn lint" under packages/components didn't raise a warning about this).

Reply:
It seems there's some other lint warning underneath (not from this PR):
```bash
events-helsinki-monorepo/packages/components (TH-1317-hide-ended-event-from-search)
$ yarn lint

events-helsinki-monorepo\packages\components\src\components\matomoWrapper\MatomoWrapper.tsx
  5:28  warning  'useState' is defined but never used  @typescript-eslint/no-unused-vars

✖ 1 problem (0 errors, 1 warning)
```